### PR TITLE
Document the new and improved review guidelines. [skip ci]

### PR DESCRIPTION
**Pull request by jpakkane**
Proposal to spread the merge work to more people. Comments welcome.

### CODE REVIEW

**Comment by trhd** [comment]
It might not be clear to all, of whom the development team consists of.

**Comment by sarum9in** [comment]
I think it's worth to mention improvements to test coverage. The case I am talking about is adding a test that passes and tests documented behavior.

**Comment by sarum9in** [comment]
And I am not sure if it belongs here, but second thing that came to my mind is about handling repositories other than https://github.com/mesonbuild/meson but under https://github.com/mesonbuild. Which group they belong to?

**Comment by jpakkane** [comment]
This is only for the main Meson repo. Adding stuff to wrapdb and related code could in theory be done by anyone with access rights. I see no reason to have a limitation on those, but if someone has, please speak up.

File: docs/markdown/Contributing.md [hunk 1]
```diff
@@ -18,6 +18,25 @@ Github](https://github.com/mesonbuild/meson/pulls). This causes them
...
+The kind of review and acceptance any merge proposal gets depends on
+the changes it contains.
+
+Documentation only changes can be accepted by any member of the
+development team who has merge rights on Github. It is also permitted
+to push doc-only fixes directly to master.
+
+The second kind of merges include refactorings that don't change any
+functionality, fixes to the CI system and bug fixes that have the
+necessary tests and don't change existing functionality. These require
```
Comment (sarum9in) [inline docs/markdown/Contributing.md]: Do you mean that tests are already present in the master or PR comes with test coverage for a bug? Might be worse clarifying.

File: docs/markdown/Contributing.md [hunk 2]
```diff
@@ -18,6 +18,25 @@ Github](https://github.com/mesonbuild/meson/pulls). This causes them
 to be run through the CI system. All submissions must pass a full CI
 test run before they are even considered for submission.
 
+## Acceptance and merging
+
+The kind of review and acceptance any merge proposal gets depends on
+the changes it contains.
+
+Documentation only changes can be accepted by any member of the
```
Comment (sarum9in) [inline docs/markdown/Contributing.md]: Might be worth clarifying if this is `doc/` subdir only or fixes of embedded help strings, comments in code, etc are accepted.

File: docs/markdown/Contributing.md [hunk 3]
```diff
@@ -18,6 +18,25 @@ Github](https://github.com/mesonbuild/meson/pulls). This causes them
...
+development team who has merge rights on Github. It is also permitted
+to push doc-only fixes directly to master.
+
+The second kind of merges include refactorings that don't change any
+functionality, fixes to the CI system and bug fixes that have the
+necessary tests and don't change existing functionality. These require
+a review from someone from the development team. Once successfully
```
Comment (sarum9in) [inline docs/markdown/Contributing.md]: If member of development team wants to do such change, do they need to get another member to review it?

**Comment by sarum9in** [comment]
Also might be worth to consider core functionality vs modules or dependencies.
E.g. adding special dependency types does not need approval from project lead. Or adding unstable_module. Something like this might make it possible to shard work across different subsystems without breaking core language.

**Comment by jpakkane** [comment]
Dependencies are ok, adding new modules is not. Even unstable ones.

**Comment by sarum9in** [comment]
I am not sure I understood your idea about reviews. Did you mean that changes in the second group have to be reviewed by someone other than contributor? Does it apply only to outside contributors or members also (member reviews other member's code)?

Listified version:
- Members:
  - Documentation only changes
  - Refactoring
  - New dependencies special cases, e.g new `foo-tool` dependency
- Leader:
  - New functionality
  - Incompatible changes (anywhere)
  - New modules
  - Stable module changes
- *arguable*
  - Unstable module changes, *probably members*
  - Additional functionality in existing dependencies, *probably leader*

I prefer listified version for very simple reason: it is more specific and we can extend it easy later if we have some use cases we are not sure about. Obviously introductory text is pretty useful also :)

File: docs/markdown/Contributing.md [hunk 4]
```diff
@@ -18,6 +18,39 @@ Github](https://github.com/mesonbuild/meson/pulls). This causes them
...
+## Acceptance and merging
+
+The kind of review and acceptance any merge proposal gets depends on
+the changes it contains. All pull requests must be reviewed and
+accepted by someone with commit rights who is not the original
+submitter. Merge requests can be roughly split into two different
+categories.
+
+The first consists of merges that don't change any functionality,
+fixes to the CI system and bug fixes that have added regression tests
```
Comment (sarum9in) [inline docs/markdown/Contributing.md]: Does #3195 fall into "bug fixes that have added regression tests"?

File: docs/markdown/Contributing.md [hunk 5]
```diff
@@ -18,6 +18,39 @@ Github](https://github.com/mesonbuild/meson/pulls). This causes them
 to be run through the CI system. All submissions must pass a full CI
 test run before they are even considered for submission.
 
+## Acceptance and merging
+
+The kind of review and acceptance any merge proposal gets depends on
+the changes it contains. All pull requests must be reviewed and
+accepted by someone with commit rights who is not the original
```
Comment (sarum9in) [inline docs/markdown/Contributing.md]: Just to clarify, this requires review of documentation changes by someone else as well, right?
In my opinion it is reasonable to allow simple non-code changes to be pushed without review,
for complex documentation changes review might be necessary.

**Comment by sarum9in** [comment]
New version looks pretty nice.

File: docs/markdown/Contributing.md (hunk 4 above)
Comment (jpakkane) [reply docs/markdown/Contributing.md]: I would say so, yes.

**Comment by jpakkane** [comment]
Updated.

**Comment by sarum9in** [comment]
LGTM

File: docs/markdown/Contributing.md [hunk 6]
```diff
@@ -18,6 +18,45 @@ Github](https://github.com/mesonbuild/meson/pulls). This causes them
...
+submitter. Merge requests can be roughly split into three different
+categories.
+
+The first one consists of MRs that only change the markdown
+documentation under `docs/markdown`. Anyone with access rights can
+push changes to these directly to master. For major changes it is
+still recommended to create a MR so other people can comment on it.
+
+The second goup consists of merges that don't change any
```
Comment (jon-turney) [inline docs/markdown/Contributing.md]: goup -> group ?

**Pull request merged**